python/calico/felix/profilerules.py

```diff
@@ -246,9 +258,11 @@ def _delete_chains(self):
         """
         # Need to block here: have to wait for chains to be deleted
         # before we can decref our ipsets.
-        self._iptables_updater.delete_chains(
-            self.iptables_generator.profile_chain_names(self.id),
-            async=False)
+        chain_names = self.iptables_generator.profile_chain_names(self.id)
+        if self._untracked:
```

Comment:
These conditionals feel like a code smell to me. Why not make the decision on raw/filter in the manager class and then only pass in the relevant iptables updater?

One way to make that work would be to adjust the profile ID to include the raw/filter bool. that'd have a couple of advantages:
- The create method would know which updater to pass in.
- The raw/filter nature would be part of the ID rather than part of the data so handling a change to that flag becomes a delete, then an add.